code-generators/market-hour-code-generator.py

```diff
@@ -218,34 +219,49 @@
                     html_file.write('''</tbody>
 </table>''')
 
-            with open(symbol_path / f"{MARKET_HOUR.HOLIDAY}.html", "w", encoding="utf-8") as html_file:
+            file_name = f"{MARKET_HOUR.HOLIDAY}.html" if not option_pages else f"{MARKET_HOUR.HOLIDAY}.php"
+            with open(symbol_path / file_name, "w", encoding="utf-8") as html_file:
                 html_file.write("<!-- Code generated by market-hour-code-generator.py -->")
                     
-                if "table" in holiday_html:
+                if option_pages:
```

Comment:
I don't know if this condition is future-proof. Maybe we should also check if the Option early closes/early opens/holidays is acutally 0 to match [how lean works](https://github.com/QuantConnect/Lean/pull/7007/files#diff-dbc7f0a9d3c57a730af34662673e7c90c7f6fc467a036bcd24d4ee331ce71eb4R281-R296)

Reply:
I switch the order of the if-condition, such that it checks if any specific content first, -> then see if it is option.